vmm/src/vm.rs

```diff
@@ -1616,6 +1720,41 @@ impl<'a> Vm<'a> {
         // Unblock all CPU threads.
         vcpu_thread_barrier.wait();
 
+        let mut signals: [i32; MAX_SIGNAL_VAL] = [0; MAX_SIGNAL_VAL];
+        let mut index = 0;
+        for sig_num in 1..MAX_SIGNAL_VAL {
+            if self.devices.signal_present[sig_num] {
+                signals[index] = sig_num as i32;
+                index += 1;
+            }
+        }
+        if index > 0 {
```

Comment:
Just test for Vec emptiness.

Reply:
Explained the action in previous reply. 
(I like the idea of replacing the array with a vector.)